extensions/vscode/lsp-client/src/extension.ts

```diff
@@ -51,12 +46,316 @@ export function activate(context: ExtensionContext) {
 
 	// Start the client. This will also launch the server
 	client.start();
-}
 
 
+	context.subscriptions.push(
+		vscode.commands.registerCommand('timeline.create', () => {
+			TimelinePanel.createOrShow();
+		})
+	);
+
+	context.subscriptions.push(
+		vscode.commands.registerCommand('timeline.refresh', () => {
+			if (TimelinePanel.currentPanel) {
+				TimelinePanel.currentPanel.refresh();
+			}
+		})
+	);
+
+	if (vscode.window.registerWebviewPanelSerializer) {
+		// Make sure we register a serializer in activation event
+		vscode.window.registerWebviewPanelSerializer(TimelinePanel.viewType, {
+			async deserializeWebviewPanel(webviewPanel: vscode.WebviewPanel, state: any) {
+				TimelinePanel.revive(webviewPanel);
+			}
+		});
+	}
+}
+
 export function deactivate(): Thenable<void> | undefined {
 	if (!client) {
 		return undefined;
 	}
 	return client.stop();
 }
+
+function getWebviewOptions(): vscode.WebviewOptions {
+	return {
+		// Enable javascript in the webview
+		enableScripts: true,
+		// And restrict the webview to not loading any content.
+		localResourceRoots: []
+	};
+}
+
+/**
+ * Manages timeline webview panels
+ */
+class TimelinePanel {
+	/**
+	 * Track the current panel. Only allow a single panel to exist at a time.
+	 */
+	public static currentPanel: TimelinePanel | undefined;
+
+	public static readonly viewType = 'timeline';
+
+	private readonly _panel: vscode.WebviewPanel;
+	private _disposables: vscode.Disposable[] = [];
+
+	public static createOrShow() {
+		const column = vscode.window.activeTextEditor
+			? vscode.window.activeTextEditor.viewColumn
+			: undefined;
+
+		// If we already have a panel, show it.
+		if (TimelinePanel.currentPanel) {
+			TimelinePanel.currentPanel._panel.reveal(column);
```

Comment:
What does this do if `column` is undefined?

Reply:
The docs say, "Shows in the current viewColumn if undefined".